File: unmodified_docker/pmm.py
from anydbver_run_tools import run_fatal, soft_params
from anydbver_common import DEFAULT_PASSWORD, logger

def deploy(node_args, node_name, net):
  params = soft_params(node_args)
  pmm_port = ""
  if "port" in params and params["port"] != "":
    pmm_port = params["port"] + ":"

  docker_run_cmd =["docker", "run", "-d", "--name={}".format(node_name),
             "-p", "{port}443".format(port=pmm_port),
             "--network={}".format(net) ]

  if "memory" in params:
    docker_run_cmd.append("--memory={}".format(params["memory"]))

  if type(params["docker-image"]) == type(True):
    docker_run_cmd.append("percona/pmm-server:{ver}".format(ver=params["version"]))
  else:
    docker_run_cmd.append(params["docker-image"])

  run_fatal(logger, docker_run_cmd, "Can't start PMM")

  run_fatal(logger,
            [
  "docker", "exec", node_name, "bash", "-c", 'sleep 30;grafana-cli --config /etc/grafana/grafana.ini --homepath /usr/share/grafana --configOverrides cfg:default.paths.data=/srv/grafana admin reset-admin-password {}'.format(DEFAULT_PASSWORD)
  ], "Can't change PMM password")

File: pkg/unmodified_docker/pmm.go
package unmodifieddocker

import (
	"log"
	"regexp"
	"strings"

	anydbver_common "github.com/zelmario/anydbver/pkg/common"
	"github.com/zelmario/anydbver/pkg/runtools"
)

func CreatePMMContainer(logger *log.Logger, namespace string, name string, cmd string, args map[string]string) {
	cmd_args := []string{
		"docker", "run",
		"--name", anydbver_common.MakeContainerHostName(logger, namespace, name),
		"-d",
		"--network", anydbver_common.MakeContainerHostName(logger, namespace, "anydbver"),
		"--hostname", anydbver_common.MakeContainerHostName(logger, namespace, name),
	}
	if mem, ok := args["memory"]; ok {
		cmd_args = append(cmd_args, "--memory="+mem)
	}

	pmmPort := "8443"
	if strings.HasPrefix(args["version"], "1.") || strings.HasPrefix(args["version"], "2.") {
		pmmPort = "443"
	}

	if _, ok := args["expose"]; ok {
		cmd_args = anydbver_common.AppendExposeParams(cmd_args, args)
	} else if port, ok := args["port"]; ok {
		cmd_args = append(cmd_args, "-p", port+":"+pmmPort)
	} else {
		cmd_args = append(cmd_args, "-p", ":"+pmmPort)
	}

	cmd_args = append(cmd_args, args["docker-image"]+":"+args["version"])

	env := map[string]string{}
	errMsg := "Error creating container"
	ignoreMsg := regexp.MustCompile("ignore this")
	runtools.RunFatal(logger, cmd_args, errMsg, ignoreMsg, true, env)

	runtools.RunFatal(logger, []string{
		"docker", "exec", anydbver_common.MakeContainerHostName(logger, namespace, name), "bash", "-c", "sleep 30;grafana-cli --config /etc/grafana/grafana.ini --homepath /usr/share/grafana --configOverrides cfg:default.paths.data=/srv/grafana admin reset-admin-password " + anydbver_common.ANYDBVER_DEFAULT_PASSWORD,
	}, "Error changing password", ignoreMsg, true, env)
}
